multimodal/src/autogluon/multimodal/utils/hpo.py

```diff
@@ -36,7 +41,8 @@ def hpo_trial(sampled_hyperparameters, predictor, checkpoint_dir=None, **_fit_ar
     if checkpoint_dir is not None:
         _fit_args["resume"] = True
         _fit_args["ckpt_path"] = os.path.join(checkpoint_dir, RAY_TUNE_CHECKPOINT)
-    predictor._fit(**_fit_args)
+    with set_torch_num_threads(num_cpus=num_cpus):
```

Comment:
What will happen if multiple trials are running in parallel? Will they have conflicts in using `num_cpus`? Is `num_cpus ` is the number of cpus of this machine?

Reply:
num_cpus is being calculated already. If 1 machine has 32 cpus and 2 trials running in parallel. Within each trial, the `num_cpus` will be 16

Reply:
Since `num_cpus` is already calculated before calling this function `hpo_trial()`, why does ray still set `torch.set_num_threads(1)`?

Reply:
I don't know why ray set that. We tell ray how much cpu to use for each tasks and it's only being used by ray for scheduling purpose. We need to control the amount of compute the task actually used. The same is being done inside tabular too.

We don't want to do a `torch.set_num_threads()` in core as resource control are specific to different models